plugin/pkg/shoot/quotavalidator/admission.go

```diff
@@ -83,15 +123,454 @@ func (h *RejectShootIfQuotaExceeded) Admit(a admission.Attributes) error {
 	if a.GetKind().GroupKind() != garden.Kind("Shoot") {
 		return nil
 	}
+	if a.GetSubresource() != "" {
+		return nil
+	}
+
 	shoot, ok := a.GetObject().(*garden.Shoot)
 	if !ok {
 		return apierrors.NewBadRequest("could not convert resource into Shoot object")
 	}
 
-	// TODO: implement quota limit check logic here
-	if false {
-		return admission.NewForbidden(a, fmt.Errorf("quota limits exceeded for %v", shoot.Name))
+	// Pass if the shoot is intended to get deleted
+	if shoot.DeletionTimestamp != nil && shoot.Annotations[common.ConfirmationDeletionTimestamp] != "" {
+		return nil
+	}
+
+	quotaReferences, err := h.getShootsQuotaReferences(*shoot)
+	if err != nil {
+		return apierrors.NewInternalError(err)
+	}
+
+	// Quotas are cumulative, means each quota must be not exceeded that the admission pass.
+	var maxShootLifetime *int
+	for _, quotaRef := range quotaReferences {
+		quota, err := h.quotaLister.
```

Comment:
Can we write these statements into one line (line breakings are not used anywhere else in the code)?

Reply:
done